src/libappimage/libappimage.c

```diff
@@ -397,49 +397,190 @@ gchar **squash_get_matching_files_install_icons_and_mime_data(sqfs* fs, char* pa
     return (gchar **) g_ptr_array_free(array, FALSE);
 }
 
-/* Loads a desktop file from squashfs into an empty GKeyFile structure.
- * FIXME: Use sqfs_lookup_path() instead of g_key_file_load_from_squash()
- * should help for performance. Please submit a pull request if you can
- * get it to work.
+
+ struct {
+     bool succeed;
+     sqfs_inode inode;
+ } typedef sqfs_lookup_result;
```

Comment:
You could've used a pointer parameter and kept the boolean return value, like we do elsewhere:

```diff 
- gboolean g_key_file_load_from_squash(sqfs *fs, char *path, GKeyFile *key_file_structure, gboolean verbose)
+ gboolean g_key_file_load_from_squash(sqfs *fs, char *path, GKeyFile *key_file_structure, sqfs_inode* inode)
```

The function could've allocated a `sqfs_inode` (or copied) and return that.

Reply:
"return parameters" are really something that I really found as a bad development idea. They just create confusion for the reader. Take for example this [function](https://github.com/AppImage/libappimage/pull/16/files#diff-70e9697814ed583654f4a7d875e018c4R518), you cannot tell here if you have to pass something in or expect something back (actually it's both you pass something and you get something). Sadly it's a practice widely spread in the C world.

This struct only add a few lines of code and increase a lot the readablity, there will be no missunderstandings here. Therefore I would like to introduce this practice in the newly written code.

Reply:
That's classic C style. Think of `(f)read` etc. People _do_ understand it, as they have to anyway.

I'd rather stay consistent with the established style instead of having so many different new structs.

You should discuss such ideas first next time, then we can decide whether to switch to new styles.

Reply:
Fixed.